sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/RecordWriterManager.java

```diff
@@ -91,6 +91,7 @@ class DestinationState {
     final Cache<PartitionKey, RecordWriter> writers;
     private final List<SerializableDataFile> dataFiles = Lists.newArrayList();
     @VisibleForTesting final Map<PartitionKey, Integer> writerCounts = Maps.newHashMap();
+    private final List<Exception> exceptions = Lists.newArrayList();
```

Comment:
Does this need concurrency controls? If not, LGTM.

Reply:
No, thread safety isn't a concern here because a RecordWriterManager instance isn't shared between threads